example_user_config.py

```diff
@@ -0,0 +1,13 @@
+from pathlib import Path
+from urllib.parse import urljoin
+
+from glados.config import Config
+
+
+config = Config()
+
+##############################################################################
+# Edit values here, based on glados/config.py defaults
+
+config.LLAMA_SERVER_EXTERNAL = True
+config.LLAMA_SERVER_BASE_URL = "http://localhost:5000/"

```

Comment:
Wouldn't it be possible to simplify this config file by exporting a global config instance here (a simple key-value pair map), and using that import in the project?

That way `user_config.py` does not need to rely on the project itself.

Or maybe go a step further and use a JSON/YAML/INI format config.

Reply:
Yeah, I think yaml or toml is probably the ideal, but I'm trying to take baby steps here to give people a sensible way to do local config without massive changes to the codebase, for now.  At the moment, I've basically moved the global vars inside an object to allow initial thoughts on variable dependency and refactoring/isolation in future.  It's difficult to see what code does if it can reference global vars at any time. However, move those vars into an object and pass the object, and then you can start to see a) what it needs; b) what it doesn't need; c) structure of what different parts need and how you might break that config object up into smaller units of just what each part of the system needs to know.  Then you can get to cleaner interfaces, clean implementations, and adding new features without breaking things because more of the code is modular.

Reply:
I think an example default config would work here, and we could gitignore YAML files to avoid accidental uploading of passwords etc

Reply:
Ok, the main branch now uses a YAML configuration for most of the parameters.

This includes the 'personality', and the URL and API keys for other backends or OpenAPI etc.